engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/age/AbstractAgeInstantBasedValidator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv.age;
+
+import java.lang.annotation.Annotation;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidator;
+
+/**
+ * Base class for all age validators that use an {@link Instant} to be compared to the age reference.
+ *
+ * @author Hillmer Chona
+ * @since 6.0.8
+ */
+public abstract class AbstractAgeInstantBasedValidator<C extends Annotation, T> implements HibernateConstraintValidator<C, T> {
+
+	protected Clock referenceClock;
+
+	protected int referenceAge;
+
+	protected boolean inclusive;
+
+	protected ChronoUnit unit;
+
+	@Override
+	public boolean isValid(T value, ConstraintValidatorContext context) {
+		// null values are valid
+		if ( value == null ) {
+			return true;
+		}
+
+		long result =  this.getCurrentAge( value ) - this.referenceAge;
```

Comment:
Done

Reply:
@marko-bekhta the method LocalDate.ofInstant( ... ) comes on Java 9, I'm no sure about to use Java 9 features.  I use next line than is  compatible with Java 8

    getInstant( value ).atZone( ZoneOffset.ofHours( 0 ) ).toLocalDate()

WDYT?

Reply:
Right... We don't want Java 9 stuff :).  I was suggesting to use `LocalDateTime` which should allow to use any of `ChronoUnit`s . Otherwise we would have the ability to do something like:
```java
class MyClass {
    @AgeMin( value = 24 * 60, unit = ChronoUnit.MINUTES )
    private LocalDate date;

    //.....
}
```
but the `add` operation would fail for such unit on `LocalDate`. Also I guess it will answers your other question about supported types. I think this constraint can have a more wider meaning. We could think of it as a constraint on how much time units have passed till now. And would allow having something like:
```java
class Exam {
    @AgeMin(value = 5, unit = ChronoUnit.MINUTES, message = "You should at least spent 5 minutes on the exam, before you can submit it")
    @AgeMax(value = 60, unit = ChronoUnit.MINUTES, message = "You cannot submit your exam as the allowed time limit have been exceeded ")
    private LocalTime examTime;
    
    // ....
}
```
or something like:
```java
class Token {
    @AgeMax(value = 10, unit = ChronoUnit.MINUTES, message = "Token expired")
    private LocalDateTime issuedAt;

    // ....
}
```